.github/workflows/ci.yml

```diff
@@ -13,10 +13,12 @@ jobs:
           cache: 'pip'
       - name: Setup enviroment
         run: pip install -r requirements-dev.txt
+      - name: Linting
+        run: make lint
       - name: Run tests + cov report
         run: make test-with-coverage
       - name: Check style
-        run: make checkstyle
+        run: make black
```

Comment:
I would combine those style checks so

```
      - name: Run tests + cov report
        run: make test-with-coverage
      - name: Check style
        run: make checkstyle
```

so effectively just leave it as is as you added lint+flake8 to checkstyle

Reply:
I think it make sense for the user to have output from black separated from the one from (pylint and flake8)

pylint and flake8 also have some warning about formatting and indentation that black will detect as error too.

But the part I like the user perceive as lint is all the warning not about indentation like: unused variables, duplicated code, too complex code, ...

Reply:
Ah, I guess the both of us had a similar idea. In #148 I actually used flake8-black which avoids having to call multiple tools and instead integrates them

Reply:
> pylint and flake8 also have some warning about formatting and indentation that black will detect as error too.

You mean part of the checks are redundant? That's certainly true. But I don't understand why then you would need to separate.

> In order of importance the code should pass …

hm, I consider black always part of "static analysis". Another reason why I suggest to just call the meta-target checkstyle in CI as well is to make sure the target actually works to prevent errors like in https://github.com/openSUSE/qem-bot/pull/147#discussion_r1363818414 :)

Reply:
ok good for me to so :-)